service/history/archival/archiver.go

```diff
@@ -260,7 +265,7 @@ func (a *archiver) archiveVisibility(ctx context.Context, request *Request, logg
 		HistoryLength:      request.HistoryLength,
 		Memo:               request.Memo,
 		SearchAttributes:   searchAttributes,
-		HistoryArchivalUri: request.HistoryURI.String(),
```

Comment:
This value is not initialized [here](https://github.com/temporalio/temporal/blob/008f056d6528142929b30aeea7a0648c2c0fef16/service/history/archival_queue_task_executor.go#L162-L201) when history archival is disabled

Reply:
Does the visibility archiver need to have the value for the historyuri?

An alternative fix would be to always fill out both URIs in the request generation method so that it never reaches here nil. Then it could presumably have the correct value instead of an empty string.

Reply:
@tristan-instacart 
> fill out both URIs

Fill them out how?
The branching for filling this out is currently set on history archival being enabled.
URI values also cannot be empty (parsing will fail).
How can we fill out the value if it were to not be available?

My understanding of this record is to record the current values for the fields and the error is that this crashes when history URI is nil (but should be empty)